mlx/backend/metal/reduce.h

```diff
@@ -16,7 +16,8 @@ void all_reduce_dispatch(
     const std::string& op_name,
     CommandEncoder& compute_encoder,
     metal::Device& d,
-    const Stream& s);
+    const Stream& s,
+    std::vector<array>& copies);
```

Comment:
That's funny that the compiler/linker didn't complain.. I guess it's not used outside `reduce.cpp`